### PR TITLE
redirecting passes

### PR DESCRIPTION
**Pull request by Javitocatral**

### CODE REVIEW

File: src/lib/assets/apple-wallet.png
Comment (zwergius) [inline src/lib/assets/apple-wallet.png:1]: Did you not have svg versions?

File: src/routes/cards/[cardsName]/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  export let data
+  const { cardData, cardNameParam } = data
+  let isSafari: boolean
+  let isIOS: boolean
+  let isAndroid: boolean
+  const oldUrl = `https://www.vermouth.nu/${cardNameParam}-vermouth-nu.pkpass`
+
+  function downloadPass() {
+    if (isIOS && isSafari) {
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:15]: we need to check this, what happens if I use chrome or firefox?

File: src/routes/cards/[cardsName]/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      if (oldUrl) {
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:16]: this is always true

File: src/routes/cards/[cardsName]/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        redirect(301, `${window.location.origin}/${cardNameParam}-vermouth-nu.pkpass`)
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:17]: don't think you are supposed to use this outside a load function

File: src/routes/cards/[cardsName]/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      }
+      window.location.href = `${window.location.origin}/${cardNameParam}-vermouth-nu.pkpass`
+    } else if (isAndroid) {
+      if (oldUrl) {
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:21]: always true

File: src/routes/cards/[cardsName]/+page.svelte [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    const { userAgent } = navigator
+    isIOS = /(iPhone|iPad|iPod)/.test(userAgent)
+    isSafari = /Safari/.test(userAgent)
+    isAndroid = /(Android)/.test(userAgent)
+    downloadPass()
+  }
+</script>
+
+<div class="download">
+  <button on:click={downloadPass}>
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:37]: this is a link not a button

File: src/routes/cards/[cardsName]/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    <img src={downloadIos} alt="download" />
+  </button>
+  <button on:click={downloadPass}>
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:40]: same

File: src/routes/cards/[cardsName]/+page.svelte [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    <img src={downloadAndroid} alt="download" />
+  </button>
+</div>
+
+<style>
+  .download {
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:46]: bad name

File: src/routes/cards/[cardsName]/+page.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
...ciI6IlRlbDoiLCJib2R5IjoiKzQ1IDI3IDUxIDEzIDQ5In0seyJpZCI6Im1haWw6IiwiaGVhZGVyIjoiTWFpbDoiLCJib2R5IjoiaW5mb0B2ZXJtb3V0aC5udSJ9LHsiaWQiOiJjdnIiLCJoZWFkZXIiOiJDVlI6IiwiYm9keSI6IjQwNjQ5Mzk1In0seyJpZCI6ImFkZHJlc3MiLCJoZWFkZXIiOiJBZGRyZXNzIiwiYm9keSI6ImMvbyBUaG9tYXMgVmFzZSBCb2xzZXRoIFxuT3ZlcmdhZGVuIE5lZGVuIFZhbmRldCA0OSBCLCBzdCAudHYgXG4xNDE0IEvDuGJlbmhhdm4gSyJ9XSwiaGV4QmFja2dyb3VuZENvbG9yIjoiI2YwYWFhOSIsImJhcmNvZGUiOnsidHlwZSI6IlFSX0NPREUiLCJ2YWx1ZSI6Imh0dHBzOi8vd3d3LnZlcm1vdXRoLm51L2NhcmRzL3Rob21hcyIsImFsdGVybmF0ZVRleHQiOiJTaGFyZSJ9LCJoZXJvSW1hZ2UiOnsic291cmNlVXJpIjp7InVyaSI6Imh0dHBzOi8vaW1hZ2VkZWxpdmVyeS5uZXQvck9UYzl0S0NUUUJjOXp0a2lCVFhfdy85MDRhYzEyNC1kZDAxLTQ0NWEtMzI2ZC0xYTEzZTM2MjUyMDAvcHVibGljIn0sImNvbnRlbnREZXNjcmlwdGlvbiI6eyJkZWZhdWx0VmFsdWUiOnsibGFuZ3VhZ2UiOiJkYS1ESyIsInZhbHVlIjoiSEVST19JTUFHRV9ERVNDUklQVElPTiJ9fX19XX0sImlhdCI6MTY5NjI1MjU3MX0.hdPNaQlKmhCNWEHchl_F64-oo-zSfI4oaaFslea3mjZcKuiJKyP5WJJcEhQ6e69ioydJgLImiYDRscpUl5Zh16gIOdGs6veAcz4TWc7R9RixRNMDeTvdl0-8oP3HOilMoWVEYEe0ylQ5E9PWhgKkHtjb8um4Tjt6UKdL9Ban4uoVNMUCMxNVzeU268RiLD61kKEF-Ejul7ZW7-sC4Zx7xmvTjl6DhEEpNw_zI5uiXNY71PEPOd_vhtHA0rJx-yGwW6etLhwDv1uiCqg98wVhGo9bulq4zD5KqywC1wr6TotKD2mw8N-HVGpHs-dwYQ9x-T7ZNpV37LSqfJz-FDUUmA',
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.ts:16]: why the extra space?

File: tsconfig.json [hunk 9]
```diff
@@ -8,12 +8,24 @@
...
+    "strict": true,
+    "preserveValueImports": true,
+    "isolatedModules": false
   },
   "paths": {
     "$lib": ["src/lib"],
     "$lib/*": ["src/lib/*"]
-  }
+  },
+  "include": [
```
Comment (zwergius) [inline tsconfig.json:19]: you added this?

File: src/routes/cards/[cardsName]/types.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+export enum CardNames {
+  Christian = 'christian',
+  Christoffer = 'christoffer',
+  Thomas = 'thomas',
+}
+
+export interface CardData {
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/types.ts:7]: what is this used for?

**Review by zwergius — CHANGES_REQUESTED** [review]

File: src/lib/assets/apple-wallet.png
Comment (Javitocatral) [reply src/lib/assets/apple-wallet.png:1]: for apple it is an svg that must be resized and google is a png just unify both png

File: src/routes/cards/[cardsName]/+page.svelte (hunk 1 above)
Comment (Javitocatral) [reply src/routes/cards/[cardsName]/+page.svelte:15]: I didn't verify it until you asked me to.

File: src/routes/cards/[cardsName]/types.ts (hunk 10 above)
Comment (Javitocatral) [reply src/routes/cards/[cardsName]/types.ts:7]: yes Record<string, CardData>

File: src/routes/cards/[cardsName]/+page.ts (hunk 8 above)
Comment (Javitocatral) [reply src/routes/cards/[cardsName]/+page.ts:16]: There is no extra space, only the one that exists to separate the cost from the function.

File: src/routes/cards/[cardsName]/+page.svelte [hunk 11]
```diff
@@ -23,9 +23,9 @@
   }
 </script>
 
-<div class="add-wallet mobile">
-  <a href={pkpass}><img class="ios" src={iosWallet} alt="button" /></a>
-  <a href={cardData.passGoogle}><img class="android" src={googleWallet} alt="button" /></a>
+<div class="add-wallet">
+  <a href={pkpass}><img class="image-button" src={iosWallet} alt="button" /></a>
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:27]: image-button is bad class name, it's not a button

alt should be more descriptive

File: src/routes/cards/[cardsName]/+page.svelte [hunk 12]
```diff
@@ -23,9 +23,9 @@
...
+  <a href={cardData.passGoogle}><img class="image-button" src={googleWallet} alt="button" /></a>
```
Comment (zwergius) [inline src/routes/cards/[cardsName]/+page.svelte:28]: same

**Review by zwergius — CHANGES_REQUESTED** [review]

**Review by zwergius — CHANGES_REQUESTED** [review]

File: src/params/name.js [hunk 13]
```diff
@@ -1,11 +1,9 @@
-import { error } from '@sveltejs/kit'
-
 const validNames = ['christian', 'christoffer', 'thomas']
 
 /** @type {import('@sveltejs/kit').ParamMatcher} */
 export function match(param) {
   if (validNames.includes(param)) {
```
Comment (zwergius) [inline src/params/name.js:5]: return validNames.includes(param)

File: src/routes/cards/[name=name]/+page.ts [hunk 14]
```diff
@@ -10,10 +10,10 @@ const cardsData: Record<string, string> = {
 }
 
 /** @type {import('./$types').PageLoad} */
-export function load({ params }) {
+export function load({ params, url }) {
   const { name } = params
   const cardData = cardsData[name]
-  const pkpass = `${import.meta.env.VITE_BASE_URL}${name}.pkpass`
+  const pkpass = `${url.origin}/${name}.pkpass`
```
Comment (zwergius) [inline src/routes/cards/[name=name]/+page.ts:16]: oleee!!

File: src/lib/components/link-wallet.svelte
Comment (zwergius) [inline src/lib/components/link-wallet.svelte:1]: delete

File: src/routes/[name]-vermouth-nu.pkpass/+page.svelte
Comment (zwergius) [inline src/routes/[name]-vermouth-nu.pkpass/+page.svelte:1]: delete

File: src/routes/cards/[name=name]/+page.svelte [hunk 15]
```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+  import { browser } from '$app/environment'
+  import googleWallet from '$lib/assets/google-wallet.svg'
+  import iosWallet from '$lib/assets/ios-wallet.svg'
+
+  export let data
+  const { cardData, pkpass } = data
+  let isIOS: boolean
```
Comment (zwergius) [inline src/routes/cards/[name=name]/+page.svelte:8]: don't declare these here, just do it inside the if

**Review by zwergius — CHANGES_REQUESTED** [review]

File: src/routes/cards/[name=name]/+page.svelte [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+  import { browser } from '$app/environment'
+  import googleWallet from '$lib/assets/google-wallet.svg'
+  import iosWallet from '$lib/assets/ios-wallet.svg'
+
+  export let data
+  const { cardData, pkpass } = data
+
+  if (browser) {
+    const { userAgent } = navigator
+    let isIOS: boolean = /(iPhone|iPad|iPod)/.test(userAgent)
```
Comment (zwergius) [inline src/routes/cards/[name=name]/+page.svelte:11]: eslint should error here - these should be const and type should be inferred

**Review by zwergius — CHANGES_REQUESTED** [review]

**Review by zwergius — APPROVED** [review]
LGTM
ok usa

**Pull request merged**